plugin/callgraph/assignment.go

```diff
@@ -0,0 +1,80 @@
+package callgraph
+
+import (
+	"slices"
+
+	"github.com/safedep/dry/utils"
+	sitter "github.com/smacker/go-tree-sitter"
+)
+
+type assignmentNode struct {
+	Namespace  string
+	AssignedTo []string
+	TreeNode   *sitter.Node
+}
+
+func newAssignmentGraphNode(namespace string, treeNode *sitter.Node) *assignmentNode {
+	return &assignmentNode{
+		Namespace:  namespace,
+		AssignedTo: []string{},
+		TreeNode:   treeNode,
+	}
+}
+
+type AssignmentGraph struct {
```

Comment:
Is this required to be exposed outside? I think this is something internal to Callgraph plugin?

Reply:
Right, currently it's not needed to be exposed, but if some utility eg. signature matcher also wants Assignment graph then we'll need it